ytsaurus_helpers.go

```diff
@@ -40,31 +45,37 @@ func doGetAllYtsaurusUsers(ctx context.Context, client yt.Client) ([]YtsaurusUse
 	var users []YtsaurusUser
 	for _, ytUser := range response {
 		var bannedSince time.Time
-		if ytUser.BannedSince != "" {
-			bannedSince, err = time.Parse(appTimeFormat, ytUser.BannedSince)
+		if bannedSinceRaw, ok := ytUser.Attrs[bannedSinceAttributeName]; ok && bannedSinceRaw != "" {
+			bannedSince, err = time.Parse(appTimeFormat, bannedSinceRaw.(string))
 			if err != nil {
 				return nil, errors.Wrapf(err, "failed to parse @banned_since. %v", ytUser)
 			}
 		}
+		var sourceUser SourceUser
+		if sourceRaw, ok := ytUser.Attrs[sourceAttributeName]; ok {
+			sourceType := AzureSourceType
+			if sourceTypeRaw, ok := ytUser.Attrs[sourceTypeAttributeName]; ok {
+				sourceType = SourceType(sourceTypeRaw.(string))
+			}
+			sourceUser, err = NewSourceUser(sourceType, sourceRaw.(map[string]any))
```

Comment:
When I've wrote this code initially I didn't succeed to make the attribute configurable and ytsaurus code knows too much about azure fields, but it is not its area of responsibility (only knowing how to communicate with ytsaurus is). Now since we can configure attribute we can fix that flaw. 

I suggest we store `map[string]any` in `YtsaurusUser.SourceData` and let app code delegate parsing it to the source (azure/ldap) if needed (I'm actually not sure if we need to parse map to AzureUser/LdapUser at all, because as far as i remember we only serialize one way AzureUser > YtsaurusUser). 
I.e. `app.buildYtsaurusUser` should either call `Source` method or maybe simpler AzureUser/LdapUser/SourceUser should have a method like `AsMap() map[string]any` and its result being written in `YtsaurusUser.SourceData` field.

That way we ensure our components are loosely coupled.